src/components/BnFileInput/BnFiIeInput.spec.ts

```diff
@@ -0,0 +1,229 @@
+import waitForExpect from 'wait-for-expect';
+import { mount } from '@vue/test-utils';
+import { defineComponent, type PropType } from 'vue';
+import { Form } from 'vee-validate';
+import BnFileInput from './BnFileInput.vue';
+
+function isRequired(val: File[] | File | undefined) {
```

Comment:
With the update of vee-validate, using this function gives a typescript error.

```
src/components/BnInput/BnInput.spec.ts:82:76 - error TS2322: Type '(val: string) => true | "This field is required"' is not assignable to type 'string | Record<string, unknown> | GenericValidateFunction<unknown> | GenericValidateFunction<unknown>[] | TypedSchema<unknown, unknown> | YupSchema<...> | undefined'.
  Type '(val: string) => true | "This field is required"' is not assignable to type 'GenericValidateFunction<unknown>'.
    Types of parameters 'val' and 'value' are incompatible.
      Type 'unknown' is not assignable to type 'string'.

82     const wrapper = mount(BnInput, { props: { name: 'input', value: false, rules: isRequired } });
```

In the end the only way I've managed to solve it is using `any` in the rules props, like this `rules: RuleExpression<any>`

Reply:
I think that meybe a better solution for this is to instance `RuleExpression`  with the type of the `modelValue`, like this: `RuleExpression<fileType>`. Because `RuleExpression` type is like this: 

```type RuleExpression<TValue> = string | Record<string, unknown> | GenericValidateFunction<TValue> | GenericValidateFunction<TValue>[] | TypedSchema<TValue> | YupSchema<TValue> | undefined;```

and the `GenericValidateFunction` is this: 

`type GenericValidateFunction<TValue = unknown> = (value: TValue, ctx: FieldValidationMetaInfo) => MaybePromise<boolean | MaybeArray<string>>;`

Reply:
ohh, if it doesn't throw type errors it probably is the correct way to handle it 👍